### PR TITLE
HBASE-24947 Addendum for HBASE-24898 to deal with Daylight Saving Time

### PR DESCRIPTION
**Pull request by bsglz**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 59s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  9s |  master passed  |
| +1 :green_heart: |  spotbugs  |   2m  3s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 28s |  the patch passed  |
| -0 :warning: |  checkstyle  |   1m  5s |  hbase-server: The patch generated 1 new + 3 unchanged - 0 fixed = 4 total (was 3)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 25s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   2m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 16s |  The patch does not generate ASF License warnings.  |
|  |   |  33m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2307 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux 430d6832ee43 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 1220a8775c |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| Max. process+thread count | 94 (vs. ulimit of 12500) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 27s |  master passed  |
| +1 :green_heart: |  compile  |   1m  9s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 48s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 44s |  hbase-server in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  4s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 34s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 39s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 137m 56s |  hbase-server in the patch passed.  |
|  |   | 166m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2307 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0633da9e4499 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 1220a8775c |
| Default Java | 2020-01-14 |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/testReport/ |
| Max. process+thread count | 4169 (vs. ulimit of 12500) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/compactions/TestCurrentHourProvider.java [hunk 1]
```diff
@@ -42,19 +40,43 @@
...
+      assertEquals(11, hour11);
+
+      // set a time represent hour 15
+      long deltaFor15 = TimeZone.getDefault().getRawOffset() - 28800000;
+      long timeFor15 = 1597907081000L - deltaFor15;
+      EnvironmentEdgeManager.injectEdge(() -> timeFor15);
+      CurrentHourProvider.tick = CurrentHourProvider.nextTick();
+      int hour15 = CurrentHourProvider.getCurrentHour();
+      if (TimeZone.getDefault().inDaylightTime(new Date(timeFor15))) {
+        hour15 = "Antarctica/Troll".equals(zoneID) ?
```
Comment (virajjasani) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/compactions/TestCurrentHourProvider.java:75]: Are we sure this is enough? No timezone other than Daylight is causing this edge case miss?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  3s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  8s |  master passed  |
| +1 :green_heart: |  compile  |   1m  1s |  master passed  |
| +1 :green_heart: |  shadedjars  |   7m  9s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 49s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 13s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 215m 21s |  hbase-server in the patch passed.  |
|  |   | 244m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2307 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0c2693e1b0f8 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 1220a8775c |
| Default Java | 1.8.0_232 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/testReport/ |
| Max. process+thread count | 3413 (vs. ulimit of 12500) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2307/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/compactions/TestCurrentHourProvider.java (hunk 1 above)
Comment (bsglz) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/compactions/TestCurrentHourProvider.java:75]: Yeah, for special datetime such as 2020-08-20 11:52:41 and 2020-08-20 15:04:00, i think so, but not sure if there are different calculation for other dates. 
Considering this case is target to test if the EnvironmentEdge take effect correctly in CurrentHourProvider, maybe it is enough?
